evm/src/cpu/jumps.rs

```diff
@@ -75,8 +74,27 @@ pub fn eval_packed_jump_jumpi<P: PackedField>(
     let is_jumpi = filter * lv.opcode_bits[0];
 
     // Stack constraints.
-    stack::eval_packed_one(lv, is_jump, stack::JUMP_OP.unwrap(), yield_constr);
-    stack::eval_packed_one(lv, is_jumpi, stack::JUMPI_OP.unwrap(), yield_constr);
+    // If stack_len != 1 if JUMP, 2 if JUMPI...
```

Comment:
One too many "if" I'm afraid 😄  Same for the recursive version

Reply:
Ah, yeah, the phrasing is weird, but it means:
If (JUMP and stack_len != 1) or (JUMPI and stack_len != 2)...